recipes/itk/all/conanfile.py

```diff
@@ -52,20 +53,29 @@ def configure(self):
         if self.settings.compiler == "apple-clang":
             raise ConanInvalidConfiguration("itk does not support this compiler")
 
+        if self.options.shared:
+            self.options["hdf5"].shared = True
+
     def requirements(self):
         self.requires("libjpeg/9d")
         self.requires("dcmtk/3.6.5")
         self.requires("double-conversion/3.1.5")
-        self.requires("eigen/3.3.7")
-        self.requires("expat/2.2.9")
-        self.requires("fftw/3.3.8")
+        self.requires("eigen/3.3.9")
+        self.requires("expat/2.2.10")
+        self.requires("fftw/3.3.9")
         self.requires("hdf5/1.12.0")
-        self.requires("icu/67.1")
-        self.requires("libtiff/4.1.0")
+        self.requires("icu/68.2")
+        self.requires("libtiff/4.2.0")
         self.requires("libpng/1.6.37")
-        self.requires("openjpeg/2.3.1")
+        self.requires("opencv/4.5.1")
+        self.requires("openjpeg/2.4.0")
         self.requires("zlib/1.2.11")
 
+    def validate(self):
+        if self.options.shared and not self.options["hdf5"].shared:
+            raise ConanInvalidConfiguration("When building a shared itk, hdf5 needs to be shared too (or not linked to by the consumer).\n"
+                                            "This is because H5::DataSpace::ALL might get initialized twice, which will cause a H5::DataSpaceIException to be thrown).")
```

Comment:
@madebr you added this check in initial PR, shouldn't it be a warning?

Reply:
If I remember correctly, the library will build, but will throw this exception when starting up.
This renders the library unusable.
Unless the library changed this behavior, I think this error should remain.